### DISCUSSION TITLE
How to run a flask app using CLI? the example dont work...

### DISCUSSION THREAD

**Discussion by aliencaocao** [cirospaciari/socketify.py#172] [Q&A] [answered]
I have the hello world Dash app:
```
from dash import Dash, html, dcc, callback, Output, Input
import plotly.express as px
import pandas as pd
import dash_bootstrap_components as dbc
from socketify import OpCode

app = Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])

app.layout = html.Div([
    html.Div(children='Hello World')
])

ws = {
    "open": lambda ws: ws.send("Hello World!", OpCode.TEXT),
    "message": lambda ws, message, opcode: ws.send(message, opcode),
    "close": lambda ws, code, message: print("WebSocket closed"),
}

# if __name__ == '__main__':
#     app.run(debug=True)
```

where app is just a Flask app.

I cannot figure out how to run it using cli:
`python -m socketify app:app --ws app:ws --host 0.0.0.0 --port 8050 --workers 8` give `socketify interface must be callable with 1 parameter def run(app: App)`. But I dont know what is run function does and even after I defined one it still give me the same error

**Comment by cirospaciari** [comment, marked as answer]
Never used dash before but looks like its not detecting it as WSGI or ASGI you can use --interface
see https://docs.socketify.dev/cli.html
